Studies/source/MvaTraining.cxx

```diff
@@ -162,43 +354,139 @@ class MVATraining {
+                        evaluation[m.first][mass_tot][type_entry.first].insert(evaluation[m.first][mass_tot][type_entry.first].end(), evaluation[m.first][entry.first][type_entry.first].begin(), evaluation[m.first][entry.first][type_entry.first].end());
+                        for (const auto& value : evaluation.at(m.first).at(entry.first).at(type_entry.first))
+                            outputBDT.at(m.first).at(mass_tot)->Fill(value);
+                    }
+                    if (entry.first.IsBackground()){
+                        for (const auto& value : evaluation.at(m.first).at(entry.first).at(type_entry.first))
+                            outputBDT.at(m.first).at(bkg)->Fill(value);
+                    }
+                }
+            }
+        }
 
-        factory->TrainAllMethods();
-        factory->TestAllMethods();
-        factory->EvaluateAllMethods();
+        std::cout<<"kolmogorov"<<std::endl;
+        std::map<std::string, std::map<SampleId,double>> kolmogorov = Kolmogorov(evaluation);
+        std::cout<<"Significativity"<<std::endl;
+        std::map<std::string, std::pair<double, double>> sign = Significativity(outputBDT);
+
+        ntuple::MvaTuple mva_tuple(outfile.get(), false);
+        auto n_trees_option = options.at("NTrees");
+        auto shrinkage_option = options.at("shrinkage");
+        auto BaggedSampleFraction_option = options.at("BaggedSampleFraction");
+        auto MaxDepth_option = options.at("MaxDepth");
+        auto MinNodeSize_option = options.at("MinNodeSize");
+        std::cout<<"options"<<std::endl;
+
+        for(const auto& point : grid) {
+            auto name = options.GetName(point)+"_"+std::to_string(args.seed());
+            size_t pos = point.at(options.GetIndex("NTrees"));
+            mva_tuple().NTrees = static_cast<UInt_t>(n_trees_option->GetNumericValue(pos));
+            pos = point.at(options.GetIndex("shrinkage"));
+            mva_tuple().shrinkage = shrinkage_option->GetNumericValue(pos);
+
+            pos = point.at(options.GetIndex("BaggedSampleFraction"));
+            mva_tuple().BaggedSampleFraction = BaggedSampleFraction_option->GetNumericValue(pos);
+            pos = point.at(options.GetIndex("MaxDepth"));
+            mva_tuple().MaxDepth = MaxDepth_option->GetNumericValue(pos);
+            pos = point.at(options.GetIndex("MinNodeSize"));
+            mva_tuple().MinNodeSize = MinNodeSize_option->GetNumericValue(pos);
+            std::cout<<"method.first"<<std::endl;
+//            mva_tuple().cut = sign[name].first;
+            std::cout<<"method.first"<<std::endl;
+//            mva_tuple().significance = sign[name].second;
+            mva_tuple().ROCIntegral = static_cast<double>(factory->GetROCIntegral(vars->loader.get(), name));
+
+            std::vector<double> ks;
+            std::vector<double> mass;
+            for (const auto& sample : kolmogorov.at(name)){
+                ks.push_back(sample.second);
```

Comment:
can do directly mva_tuple().KS_mass.push_back(sample.second)
same for all other vectors.